techniques/jobScheduling/jobScheduler/3.0/job_scheduler.st

```diff
@@ -0,0 +1,118 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+
+bundle agent job_scheduler_RudderUniqueID
+{
+  vars:
+
+    &JOB_COMMAND:{command |"job_scheduler_command[&i&]" string => "&command&";
+}&
+    &JOB_TIMEOUT:{timeout |"job_scheduler_timeout[&i&]" string => "&timeout&";
+}&
+    &JOB_KEPTCODES:{keptcodes |"job_scheduler_keptcodes[&i&]" string => "&keptcodes&";
+}&
+    &JOB_REPAIRCODES:{repaircodes |"job_scheduler_repaircodes[&i&]" string => "&repaircodes&";
+}&
+    &JOB_ERRORCODES:{errorcodes |"job_scheduler_errorcodes[&i&]" string => "&errorcodes&";
+}&
+    &JOB_TIME_MIN:{mintime |"job_scheduler_time_min[&i&]" string => "&mintime&";
+}&
+    &JOB_TIME_MAX:{maxtime |"job_scheduler_time_max[&i&]" string => "&maxtime&";
+}&
+    &TRACKINGKEY:{uuid |"job_scheduler_uuids[&i&]" string => "&uuid&";
+}&
+
+      "iterator"                               slist => getindices("job_scheduler_uuids");
+
+      "job_scheduler_canon_name[${iterator}]" string => canonify("${job_scheduler_command[${iterator}]}");
+
+
+      
+      "kept_returncodes_${iterator}"          slist => splitstring("${job_scheduler_keptcodes[${iterator}]}",",","128");
+
+      "repaired_returncodes_${iterator}"      slist => splitstring("${job_scheduler_repaircodes[${iterator}]}",",","128");
+
+      "error_returncodes_${iterator}"         slist => splitstring("${job_scheduler_errorcodes[${iterator}]}",",","128");
+
+
+    pass1::
+      "max_execution_delay_hours_real_${iterator}" string => eval("${job_scheduler_time_max[${iterator}]}-${job_scheduler_time_min[${iterator}]}", "math", "infix"),
+                                               ifvarclass => "min_time_before_max_time_${iterator}";
+
+      "max_execution_delay_hours_real_${iterator}" string => eval("24-${job_scheduler_time_min[${iterator}]}+${job_scheduler_time_max[${iterator}]}", "math", "infix"),
+                                               ifvarclass => "!min_time_before_max_time_${iterator}";
+
+      "max_execution_delay_hours_${iterator}" string => format("%d", "${max_execution_delay_hours_real_${iterator}}");
+
+  classes:
+      # we need to check if min time is greater than max time or not
+      "min_time_before_max_time_${iterator}" expression => islessthan("${job_scheduler_time_min[${iterator}]}", "${job_scheduler_time_max[${iterator}]}");
+
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+
+  methods:
+    pass2::
+      # check if the job should run
+      "schedule_${iterator}" usebundle => 
+         schedule_simple("${job_scheduler_canon_name[${iterator}]}_RudderUniqueID", "${g.agent_run_interval}", 0, "${max_execution_delay_hours_${iterator}}", "0", "${job_scheduler_time_min[${iterator}]}", "0", "0", "24", "0", "nodups");
```

Comment:
We should have a fallback on 5 mintues in case system techniques are not up to date.